commons-statistics-descriptive/src/main/java/org/apache/commons/statistics/descriptive/Variance.java

```diff
@@ -0,0 +1,233 @@
+ * <code>clear()</code> method, it must be synchronized externally.
+ *
+ * <p>However, it is safe to use <code>accept()</code> and <code>combine()</code>
+ * as <code>accumulator</code> and <code>combiner</code> functions of
+ * {@link java.util.stream.Collector Collector} on a parallel stream,
+ * because the parallel implementation of {@link java.util.stream.Stream#collect Stream.collect()}
+ * provides the necessary partitioning, isolation, and merging of results for
+ * safe and efficient parallel execution.
+ *
+ * @since 1.1
+ */
+public abstract class Variance implements DoubleStatistic, DoubleStatisticAccumulator<Variance> {
+
+    /**
+     * Create a Variance instance.
+     */
+    Variance() {
+        // No-op
+    }
+
+    /**
+     * Creates a {@code Variance} implementation which does not store the input value(s) it consumes.
+     *
+     * <p>The result is <code>NaN</code> if:
+     * <ul>
+     *     <li>no values have been added,</li>
+     *     <li>any of the values is <code>NaN</code>, or</li>
+     *     <li>an infinite value of either sign is encountered</li>
+     * </ul>
+     *
+     * @return {@code Variance} implementation.
+     */
+    public static Variance create() {
+        return new StorelessSampleVariance();
+    }
+
+    /**
+     * Returns a {@code Variance} instance that has the variance of all input values, or <code>NaN</code>
+     * if:
+     * <ul>
+     *     <li>the input array is empty,</li>
+     *     <li>any of the values is <code>NaN</code>, or</li>
+     *     <li>an infinite value of either sign is encountered</li>
+     * </ul>
+     *
+     * <p>Note: {@code Variance} computed using {@link Variance#accept Variance.accept()} may be different
+     * from this variance.
+     *
+     * <p>See {@link Variance} for details on the computing algorithm.
+     *
+     * @param values Values.
+     * @return {@code Variance} instance.
+     */
+    public static Variance of(double... values) {
+        final double mean = Mean.of(values).getAsDouble();
+        double accum = 0.0;
+        double dev;
+        double accum2 = 0.0;
+        double squaredDevSum;
+        for (final double value : values) {
+            dev = value - mean;
+            accum += dev * dev;
+            accum2 += dev;
+        }
+        final double accum2Squared = accum2 * accum2;
+        final long n = values.length;
+        // The sum of squared deviations is accum - (accum2 * accum2 / n).
+        // To prevent squaredDevSum from spuriously attaining a NaN value
+        // when accum2Squared (which implies accum is also infinite) is infinite, assign it
+        // an infinite value which is its intended value.
+        if (accum2Squared == Double.POSITIVE_INFINITY) {
+            squaredDevSum = Double.POSITIVE_INFINITY;
+        } else {
+            squaredDevSum = accum - (accum2 * accum2 / n);
+        }
+        return StorelessSampleVariance.create(squaredDevSum, mean, n, accum2 + (mean * n));
```

Comment:
I started reading the paper but ran out of time to verify the code here, but from the javadocs I think it is based on this one?

![image](https://github.com/apache/commons-statistics/assets/304786/1650255a-2e62-47b1-8772-3cd3abca6f69)

Reply:
West's algorithm is used in the `accept(double d)` method for updating the value of the statistic.
The two-pass algorithm used in the `of(double... values)` method is :

![image](https://github.com/apache/commons-statistics/assets/129569933/f1c9eb96-2926-4a5d-a61b-c14a5718c78a)
This formula is from the paper/report titled - [Algorithms for computing the sample variance : analysis and recommendations](https://www.cs.yale.edu/publications/techreports/tr222.pdf)

Reply:
Thanks for the explanation. I had a look at `SumOfSquaredDeviations#accept` and it looks correct to me!